frontend/src/editor/index.tsx

```diff
@@ -22,10 +31,53 @@ function App() {
 	// Since this is a list, a useState would have worked as well
 	const selectionChangeHandlers = useRef<(() => void)[]>([]);
 
+	// Add word count state for demo mode
+	const [wordCount, setWordCount] = useState<number>(0);
+	// Add state to track when word count exceeds limit
+	const [isOverLimit, setIsOverLimit] = useState<boolean>(false);
+
 	const handleSelectionChange = () => {
 		selectionChangeHandlers.current.forEach(handler => handler());
 	};
 
+	// Add effect to prevent input when word count exceeds limit (only in demo mode)
+	useEffect(() => {
+		if (!isDemo || !wordLimit) return;
+
+		// Find editor element, accounting for different class names in demo vs regular mode
+		const editorSelector = isDemo ? `.${classes.demoeditor}` : `.${classes.editor}`;
+		const editorElement = document.querySelector(editorSelector);
+		if (!editorElement) return;
```

Comment:
This could be simplified a lot. From least to most changes:

1. Using a React ref instead of the query
2. Using React's oninput handler (might not work for Lexical)
3. Hooking the doc context to trim off anything past the word (character?) limit
4. Hooking Lexical's state change to trim it.

Option 3 is probably the sweet spot.